extension.js

```diff
@@ -272,6 +289,38 @@ function toggleWindow() {
 	}
 }
 
+/*
+Javascript Object is implemented using a HashTable (https://stackoverflow.com/a/24196259).
+So to enable faster checking for players, 'players' should be implemented as an Object instead of an Array. Then
+we can just iterate through the current open windows and check if that window is in players. Thus O(n) instead
+of O(n^2) of checking if the window is in the players array. The Object should map the 'window_actor.meta_window.wm_class'
+(i.e. what is given after mapping a windowActor to its name) to the name that is used to identify the path in the dbus-send
+command.
+*/
+const players = {
+	amarok: 'amarok',			// not sure if works
+	vvave: 'vvave',				// not sure if works
+	elisa: 'elisa',
+	juk: 'juk',
+	plasma_media_center: 'plasma-media-center', 			  // not sure if works
+	plasma_browser_integration: 'plasma-browser-integration', // not sure if works
+	vlc: 'vlc',
+	Spotify: 'spotify',
+	Clementine: 'clementine',
+	Rhythmbox: 'rhythmbox',
+}
+
+function getPlayer() {
+	// get the names(? - is it the names?) of the current open windows
+	let windowActors = global.get_window_actors();
+	let windowNames = windowActors.map(w => w.get_meta_window().get_wm_class());
+
+	for (let windowName of windowNames) {
+		if (players[windowName])
+			return players[windowName];
+	}
+}
```

Comment:
What about players that don't have windows?  E.g., https://github.com/mavit/slimpris2/

It make more sense to ask DBus for a list of available players.  E.g.,

```bash
dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.ListNames | grep -E '^ +string "org\.mpris\.MediaPlayer2\.'
```